src/dllmain.cpp

```diff
@@ -0,0 +1,43 @@
+// dllmain.cpp : Defines the entry point for the DLL application.
+#include <windows.h>
+
+BOOL APIENTRY DllMain( HMODULE hModule,
```

Comment:
`BOOL` return value...
`BOOL` is a typedef that arrives from `<windows.h>`.
For an API that is cross-platform, it will cause undefined symbol compile errors when used on say Mac.
A cross-platform return type would have to be something like `int` instead.

That being said: `dllmain.cpp` is "windows specific".
so this is the exception where `BOOL` is fine, because your Mac version will have a completely different source file that is used for it's "shared object (.so, instead of .dll)" entry point.